astra/src/main/java/com/slack/astra/metadata/search/SearchMetadataStore.java

```diff
@@ -62,4 +71,66 @@ public CompletionStage<String> updateAsync(SearchMetadata metadataNode) {
   public void updateSync(SearchMetadata metadataNode) {
     throw new UnsupportedOperationException("Updates are not permitted for search metadata");
   }
+
+  @Override
+  public SearchMetadata getSync(String partition, String path) {
+    try {
+      LOG.info("GOT search metadata");
+      return super.getSync(partition, path);
+    } catch (Exception e) {
+      LOG.error("Error getting search metadata", e);
+      return legacyStore.getSync(path);
+    }
+  }
+
+  @Override
+  public void deleteSync(SearchMetadata metadataNode) {
+    try {
+      super.deleteSync(metadataNode);
+    } catch (Exception e) {
```

Comment:
In some cases you log an error, some you do not. You probably should log the fact that all of these are falling back to the legacy store?

Reply:
Is there a case where it would only need to delete from the legacy store?